proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/saml/HubResponseTranslator.java

```diff
@@ -119,39 +115,38 @@ private static String getMappedStatusCode(VspScenario vspScenario) {
     }
 
     private static String getCombineFirstAndMiddleNames(HubResponseContainer hubResponseContainer) {
-        final List<Attribute<String>> firstNames = hubResponseContainer.getAttributes().getFirstNames();
-        final List<Attribute<String>> middleNames = hubResponseContainer.getAttributes().getMiddleNames();
-
-        final String firstNamesCombined = combineAttributeValues(getValidAttributes(firstNames));
-
-        if (firstNamesCombined.isEmpty()) {
-            throw new HubResponseTranslationException("No verified current first name present");
+        var firstNames = hubResponseContainer.getAttributes().getFirstNames();
+        var validFirstNames = firstNames.getValidAttributes();
+        if (validFirstNames.isEmpty()) {
+            throw new HubResponseTranslationException("No verified current first name present: " + firstNames.createAttributesMessage());
         }
-
-        final String middleNamesCombined = combineAttributeValues(getValidAttributes(middleNames));
-
-        return middleNamesCombined.isEmpty() ? firstNamesCombined : firstNamesCombined + " " + middleNamesCombined;
+        List<Attribute<String>> validMiddleNames = hubResponseContainer.getAttributes().getMiddleNames().getValidAttributes();
+        validFirstNames.addAll(validMiddleNames);
+        return combineStringAttributeValues(validFirstNames);
     }
 
     private static String getCombinedSurnames(HubResponseContainer hubResponseContainer) {
-        return Optional.of(combineAttributeValues(getValidAttributes(hubResponseContainer.getAttributes().getSurnames())))
-                .filter(s -> !s.isEmpty())
-                .orElseThrow(() -> new HubResponseTranslationException("No verified current surname present"));
+        var surnames = hubResponseContainer.getAttributes().getSurnames();
+        var validSurnames = surnames.getValidAttributes();
+        if (validSurnames.isEmpty()) {
+            throw new HubResponseTranslationException("No verified current surname present: " + surnames.createAttributesMessage());
+        }
+        return combineStringAttributeValues(surnames.getValidAttributes());
     }
 
     private static String getLatestValidDateOfBirth(HubResponseContainer hubResponseContainer) {
-        return getValidAttributes(hubResponseContainer.getAttributes().getDatesOfBirth())
-                .stream()
+        var datesOfBirth = hubResponseContainer.getAttributes().getDatesOfBirth();
+        var validDatesOfBirth = datesOfBirth.getValidAttributes();
+        if (validDatesOfBirth.isEmpty()) {
+            throw new HubResponseTranslationException("No verified current date of birth present: " + datesOfBirth.createAttributesMessage());
+        }
+        return validDatesOfBirth.stream()
```

Comment:
This stuff potentially more belongs in the other ticket i.e. https://govukverify.atlassian.net/browse/EID-1750 – doesn't seem to relate to this ticket? Seems like the behaviour is unchanged though so I guess it doesn't matter.

(Out of interest, do we just not like the `orElseThrowThing` or is there a particular reason for this change? –it was new to me anyway, I would have always written it the way you now have it :-) )

Reply:
`validDatesOfBirth.isEmpty()` is a method on a list - not an Optional. If it were an Optional we could do something like orElseThrow